tests/fixture_valid.js

```diff
@@ -80,13 +80,55 @@ module.exports.checkFixture = function checkFixture(fixture, usedShortNames=[])
       }
 
       for (const ch of mode.channels) {
-        if (!(ch in fixture.availableChannels) && ch !== null) {
-          result.errors.push({
-            description: `channel '${ch}' referenced from mode '${modeShortName}' (#${i}) but missing.`,
-            error: null
-          });
-        }
         usedChannels.push(ch);
+
+        if (ch !== null) {
+          if (!(ch in fixture.availableChannels) && ch !== null) {
+            result.errors.push({
+              description: `channel '${ch}' referenced from mode '${modeShortName}' (#${i}) but missing.`,
+              error: null
+            });
+          }
+          else {
+            switch (fixture.availableChannels[ch].type) {
```

Comment:
switch / case does not really make sense for only two branches. Please use if / else if here.